Modules/gcmodule.c

```diff
@@ -1530,8 +1535,13 @@ ensure_disabled__enter__method(ensure_disabled_object *self, PyObject *args)
     PyGILState_STATE gstate = PyGILState_Ensure();
     ++_PyRuntime.gc.disabled_threads;
     self->previous_gc_state = _PyRuntime.gc.enabled;
-    gc_disable_impl(NULL);
+
+    PyObject *ret = gc_disable_impl(NULL);
     PyGILState_Release(gstate);
```

Comment:
@serhiy-storchaka Do you understand this `PyGILState_Release` call? Isn't it unsafe to execute Python/CAPI code after it?

Reply:
here and below, the 

```
    PyGILState_Release(gstate);
    if (ret == NULL) {
        return NULL;
    }
    Py_DECREF(ret);
```

blocks should be replaced with

```
    Py_XDECREF(ret);
    PyGILState_Release(gstate);
    if (ret == NULL) {
        return NULL;
    }
```

The GIL must be held for the decref.

Reply:
But this is a Python API!  So we release the gil inside `with ensure_disabled`. Why?

Reply:
I didn't review the initial changes and know nothing about the correctness of other code. I'll check it carefully after beta1.

Reply:
IMO this is just plain wrong.